File: Java/_10828.java
package Algorithm.BaekJoon;

import java.io.*;
import java.util.*;

// 스택
public class _10828 {

    public static void main(String[] args) throws IOException{

        Stack<Integer> stack = new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        while(N-- > 0){

            String[] line = br.readLine().split(" ");

            if(line[0].equals("push")){
                stack.push(Integer.parseInt(line[1]));
            } else if(line[0].equals("pop")){
                if(!stack.isEmpty()){
                    System.out.println(stack.pop());
                } else{
                    System.out.println("-1");
                }
            } else if(line[0].equals("size")){
                System.out.println(stack.size());
            } else if(line[0].equals("empty")){
                if(stack.isEmpty()){
                    System.out.println("1");
                } else{
                    System.out.println("0");
                }
            } else if(line[0].equals("top")){
                if(!stack.isEmpty()){
                    System.out.println(stack.peek());
                } else{
                    System.out.println("-1");
                }
            }
        }
    }
}

File: Python/10828.py
import sys

f = lambda : sys.stdin.readline().rstrip()

N = int(f())

stack = list()
for _ in range(N):
    line = f().split()
    if line[0] == "push":
        stack.append(int(line[1]))
    elif line[0] == "pop":
        if len(stack) == 0:
            print("-1")
        else:
            print(stack.pop())
    elif line[0] == "size":
        print(len(stack))
    elif line[0] == "empty":
        if len(stack) == 0:
            print("1")
        else:
            print("0")
    elif line[0] == "top":
        if len(stack) == 0:
            print("-1")
        else:
            print(stack[-1])

